docs/content/technical/build_process.md

```diff
@@ -0,0 +1,115 @@
+---
+title: 'Build process information'
+weight: 35
+---
+
+## 1. Introduction
+
+NDS ROMs are complicated. They have an ARM9 binary, an ARM7 binary, an optional
+filesystem image, and a ROM header with information about where to locate all
+the parts of the ROM, as well as other information like the game title and icon.
+
+You can get more information about it [in GBATEK](https://problemkaputt.de/gbatek.htm#dscartridgeheader).
+
+This document explains the way BlocksDS builds each component and assembles the
+final ROM.
+
+## 2. ARM7 and ARM9 binaries
+
+This part is pretty standard. The developer provides source code files that
+are compiled and linked into an `.elf` file (one per CPU). The source code files
+may contain code or data as requested by the developer. This code and data is
+placed into different sections as with any other `.elf` file:
+
+- `text`: Code.
+- `rodata`: Constant data.
+- `data`: Initialized variables.
+- `bss`: Uninitialized variables (they start as 0).
```

Comment:
There are a few other notable sections provided by our link scripts:

- ITCM (`.vectors` `.itcm`)/DTCM (`.dtcm` `.sbss`) sections
- Secure area (`.secure`) - space reserved for the secure area, which DS/DSi homebrew does not encrypt or use